runtime-modules/content/src/nft/mod.rs

```diff
@@ -482,4 +461,34 @@ impl<T: Trait> Module<T> {
         );
         Ok(())
     }
+
+    pub(crate) fn build_royalty_payment_for_channel(
+        channel_id: T::ChannelId,
+        creator_royalty: Option<Royalty>,
+    ) -> Option<(Royalty, T::AccountId)> {
+        // payment is none if there is no royalty
+        if let Some(royalty) = creator_royalty {
+            // payment is none if creator channel has been deleted
+            if let Ok(channel) = Self::ensure_channel_exists(&channel_id) {
```

Comment:
This is a false check, no one can call `build_royalty_payment_for_channel` this if channel has been deleted?? 
Perhaps you can isntead pass channel istelf as input or some other thing to sidestep.

Reply:
fixed here 832bf2ebf543493f7aa93d45a2acf55648695761